core/src/solution_submission.rs

```diff
@@ -297,40 +297,38 @@ fn convert_cancel_result(result: Result<(), NoopTransactionError>) -> SolutionSu
     }
 }
 
-trait IsNonceError {
-    fn is_nonce_error(&self) -> bool;
+trait IsTransactionError {
+    /// Is this an error with the transaction itself instead of an evm related error.
+    fn is_transaction_error(&self) -> bool;
 }
 
-impl IsNonceError for ExecutionError {
-    fn is_nonce_error(&self) -> bool {
-        // This is the error as we've seen it on openethereum nodes.
+impl IsTransactionError for ExecutionError {
+    fn is_transaction_error(&self) -> bool {
+        // This is the error as we've seen it on openethereum nodes. The code and error messages can
+        // be found in openethereum's source code in `rpc/src/v1/helpers/errors.rs`.
         // TODO: check how this looks on geth and infura. Not recognizing the error is not a serious
         // problem but it will make us sometimes log an error when there actually was no problem.
-        match self {
-            ExecutionError::Web3(Web3Error::Rpc(RpcError { code, message, .. }))
-                if code.code() == -32010
-                    && message.find("Transaction nonce is too low.").is_some() =>
-            {
-                true
-            }
-            _ => false,
-        }
+        let code = match self {
+            ExecutionError::Web3(Web3Error::Rpc(RpcError { code, .. })) => code.code(),
+            _ => return false,
+        };
+        code == -32010
     }
 }
 
-impl IsNonceError for Result<(), MethodError> {
-    fn is_nonce_error(&self) -> bool {
+impl IsTransactionError for Result<(), MethodError> {
+    fn is_transaction_error(&self) -> bool {
```

Comment:
naming-nit: This implementation seems very specific to OpenEthereum, so maybe this trait should be (or call internally) a `is_open_ethereum_transaction_error`?